ctapipe/image/extractor.py

```diff
@@ -1032,31 +1035,36 @@ def _apply_second_pass(
         # linear fit of pulse time vs. distance along major image axis
         # using only the main island surviving the preliminary
         # image cleaning
-        timing = timing_parameters(camera_geometry, image_2, pulse_time_1stpass, hillas)
+        timing = timing_parameters(
+            geom=camera_geometry,
+            image=image_2,
+            peak_time=pulse_time_1stpass,
+            hillas_parameters=hillas,
+            cleaning_mask=mask_biggest,
```

Comment:
this was the main bug, missing the cleaning_mask option

Reply:
thanks for having spotted this!

Reply:
Why zero out the image at all? All algorithms should support passing either masked image / geometries or image and mask

Reply:
agreed - I didn't make that change here yet since i wanted first to show the minimal bug fix, but it would be faster/easier to just slice the geometry as is done in ImageProcessor.